src/Infrastructure/BucketTesting/BucketSelector.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace WMDE\Fundraising\Frontend\Infrastructure\BucketTesting;
+
+class BucketSelector {
+
+	private $campaigns;
+ 	private $cookie;
+ 	private $urlParameters;
+
+	public function __construct( CampaignCollection $campaigns, array $cookie = [], array $urlParameters = [] ) {
```

Comment:
Let's get rid of the constructor parameters and setter and make the url and cookie variables parameters for `selectbuckets`.